flink-connectors/flink-connector-aws-kinesis-firehose/src/main/java/org/apache/flink/connector/firehose/sink/KinesisFirehoseSinkWriter.java

```diff
@@ -96,11 +96,7 @@ private static FirehoseAsyncClient createFirehoseClient(
                     RESOURCE_NOT_FOUND_EXCEPTION_CLASSIFIER,
                     getSdkClientMisconfiguredExceptionClassifier());
 
-    // deprecated, use numRecordsSendErrorsCounter instead.
-    @Deprecated private final Counter numRecordsOutErrorsCounter;
-
-    /* A counter for the total number of records that have encountered an error during put */
-    private final Counter numRecordsSendErrorsCounter;
```

Comment:
Why remove it? `numRecordsOutErrorsCounter` is deprecated but still contains the errors counter.

Reply:
They are using different counter: https://github.com/apache/flink/blob/1902699cddcafcfabf9166d4221caa3545b60943/flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/InternalSinkWriterMetricGroup.java#L48

Reply:
I've also modified `InternalSinkWriterMetricGroup` to make these two metrics share the same counter:

https://github.com/apache/flink/pull/21065/files#diff-6ecdbb02f8667509442f88d7d251838f6388b0919df9390867f987bcc6423678R46-R48

Reply:
I'm not sure if I understand your argument. Please correct me if I'm wrong. 

Firstly, we need to keep only one counter variable (no matter `sendCounter` or `outCounter`) in sink writer implementations, as these two variables are pointing to the same underlying counter actually, and `InternalSinkWriterMetricGroup` registers the underlying counter twice in the registry with two different metric names `out` and `send`. Keeping two variables and increasing both of them will finally count twice towards the same underlying counter. 

Then about which variable to remove. I think what we are arguing with is just the name of the variable in sink writer's implementation. From the user's perspective, the result is the same no matter whether the variable name is `sendCounter` or `outCounter` or `fooCounter`: the sink will register two metrics `numXXXSend` and `numXXXOut` having the same value. I think it's better to respect the original definition in FLIP-33 to use `out` to name variables for consistency.